### DISCUSSION TITLE
Can ORT be used to enrich the existing SPDX file ?

### DISCUSSION THREAD

**Discussion by dineshr93** [oss-review-toolkit/ort#8608] [Q&A]
In Doc, ORT can generate spdx but If there is half baked spdx (with only deps info) can ORT be used consume spdx and enrich with license and copyright data?

**Comment by sschuberth** [comment]
In theory, that should be possible by using the "half baked" SPDX file as input to the [SpdxDocumentFile](https://github.com/oss-review-toolkit/ort/blob/main/plugins/package-managers/spdx/src/main/kotlin/SpdxDocumentFile.kt) analyzer, running the scanner step, and then the reporter to output a new SPDX file which should contain everything from the original SPDX file plus scanner results.

In practice, though, you'll mostly likely run into all kinds of issues WRT different representation of semantically the same thing in different SPDX syntax, as SPDX in not very concise in this regard. But feel free to give it a try and share your experiences.
